src/nix/develop.cc

```diff
@@ -246,6 +246,7 @@ struct Common : InstallableCommand, MixProfile
         "NIX_LOG_FD",
         "NIX_REMOTE",
         "PPID",
+        "SHELL",
```

Comment:
Should we instead set `$SHELL` to the interactive bash we fetch from `nixpkgs`? that way most stuff will just work as expected since it will still point to bash